### PR TITLE
Fix CustomerAddress.all

### PR DESCRIPTION
**Pull request by mllemango**
### WHY are these changes introduced?
Some resources such as `CustomerAddress` can have multiple names when it comes to parsing the response Shopify returns. In this case, CustomerAddress will use `customer_address` most of the time, but it can also use `addresses`. 

We should allow multiple resourceNames for this reason

Raised by https://github.com/Shopify/shopify-api-js/issues/875 and https://github.com/Shopify/shopify-api-js/issues/384

## Type of change

- [x] Patch: Bug (non-breaking change which fixes an issue)
- [ ] Minor: New feature (non-breaking change which adds functionality)
- [ ] Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have used `yarn changeset` to create a draft changelog entry (do NOT update the `CHANGELOG.md` file manually)
- [x] I have added/updated tests for this change
- [ ] ~I have documented new APIs/updated the documentation for modified APIs (for public APIs)~

### CODE REVIEW

File: rest/base.ts [hunk 1]
```diff
@@ -212,19 +212,22 @@ export class Base {
...
-        [],
-      );
-    }
-
-    if (data[this.resourceName]) {
-      return [this.createInstance<T>(session, data[this.resourceName])];
-    }
+    let instances: T[] = [];
+    this.resourceNames.forEach((resourceName, index) => {
+      if (data[this.pluralNames[index]] || Array.isArray(data[resourceName])) {
```
Comment (paulomarg) [inline rest/base.ts:217]: This scares me a little bit 😄 - should we consider changing `resourceNames` so it's something like

```ts
resourceNames = [
  {singular: 'resource', plural: 'resources'},
  {singular: 'resource2', plural: 'resources2'},
]
```

**Review by paulomarg — COMMENTED** [review]
My comment is more of a nit, but overall this LGTM.

File: rest/base.ts (hunk 1 above)
Comment (mllemango) [reply rest/base.ts:217]: yeah we can totally do that

File: lib/clients/rest/__tests__/resources/fake-resource-with-custom-prefix.ts [hunk 2]
```diff
@@ -10,12 +10,9 @@ interface FakeResourceWithCustomPrefixFindArgs {
...
   public static apiVersion = LATEST_API_VERSION;
-  protected static resourceNames: string[] = [
-    'fake_resource_with_custom_prefix',
-  ];
-
-  protected static pluralNames: string[] = [
-    'fake_resource_with_custom_prefixes',
+  protected static resourceNames: {[key: string]: string}[] = [
+    {singular: 'fake_resource_with_custom_prefix'},
+    {plural: 'fake_resources_with_custom_prefixes'},
```
Comment (paulomarg) [inline lib/clients/rest/__tests__/resources/fake-resource-with-custom-prefix.ts:15]: I feel like these should be in the same object?

File: rest/base.ts [hunk 3]
```diff
@@ -57,8 +57,8 @@ export class Base {
   public static config: ConfigInterface;
 
   public static apiVersion: string;
-  protected static resourceNames: string[] = [];
-  protected static pluralNames: string[] = [];
+  protected static resourceNames: {[key: string]: string}[] = [];
```
Comment (paulomarg) [inline rest/base.ts:60]: We should probably create an interface for this so we can name the `singular` and `plural` fields.

**Review by paulomarg — APPROVED** [review]
Just nits!

**Pull request merged**